src/Dependency/DependencyResolver.php

```diff
@@ -176,7 +178,10 @@ public function resolveDependencies(\PhpParser\Node $node, Scope $scope): array
 			foreach ($exprType->getIterableValueType()->getReferencedClasses() as $referencedClass) {
 				$this->addClassToDependencies($referencedClass, $dependenciesReflections);
 			}
-		} elseif ($node instanceof Array_) {
+		} elseif (
+			$node instanceof Array_
+			&& $this->considerArrayForCallableTest($node)
```

Comment:
@ondrejmirtes I was implementing your suggestion and I don't think it will fix the issue in our case, since there's the indirection via `self::`.

```
class Foobar
{
    self::FOO = 'foo';
    self::FOO = 'bar';
    self::FOOBAR = [self::FOO, self::BAR];
}
```

The only way to fix it by looking at the AST that I can see is to only allow `::class`, which I agree is too strict.